src/main/java/com/auth0/net/RateLimitInterceptor.java

```diff
@@ -67,24 +67,24 @@ public Response intercept(@NotNull Chain chain) throws IOException {
             .withMaxRetries(maxRetries)
             .withBackoff(INITIAL_INTERVAL, MAX_INTERVAL, ChronoUnit.MILLIS)
             .withJitter(JITTER)
-            .handleResultIf(response -> {
-                if (response.code() == 429) {
-                    response.close();
-                    return true;
-                }
-                return false;
-            });
+            .handleResultIf(response -> response.code() == 429);
 
         // For testing purposes only, allow test to hook into retry listener to enable verification of retry backoff
         if (retryListener != null) {
             retryPolicy.onRetry(retryListener);
         }
 
         try {
+            return Failsafe.with(retryPolicy).get((context) -> {
+                // ensure response of last recorded response prior to retry is closed
+                if (context.getLastResult() != null) {
```

Comment:
instead of closing the response in the predicate above as before, close it if their is a last recorded (retried) retry. If the retries have been exceeded, the execution will be canceled and the final response will be open and able to be consumed.

Reply:
This looks good, just wondering how the final response will be closed once the body is consumed. Searched the codebase and couldn't find any other usage of close and okhttp seems to [insist](https://github.com/square/okhttp/blob/cf2a40bdb997c359c0516941015181958f4381df/okhttp/src/main/kotlin/okhttp3/Call.kt#L42-L55) closing the body.

This would be applicable for other response body as well right?

Reply:
For sync requests, it's closed using the try-with-resources block here, and the compiler will insert a close call in the finally block. But for the async case, we do need to close manually; good catch. Will push a new commit to fix.